### DISCUSSION TITLE
Error reading `.csv` file in serverless function

### DISCUSSION THREAD

**Discussion by BrianSHenderson** [vercel/community#267] [Help] [answered]
When running locally everything works well however when I deploy to Vercel I get an error:
```
Uncaught invalid stored block lengths
```
The code I'm using is:
```
                  const fileReader = new FileReader();
                  fileReader.onload = function (e) {
                    const oFile = XLSX.read(new Uint8Array(e.target.result), {
                      type: "array",
                      cellDates: true,
                      cellStyles: false,
                    });
                }
```
This is using SheetJS to open an Excel file.
Any ideas?

**Comment by leerob** [comment, marked as answer]
Hey! It looks like you're trying to read a local .csv file from the File System. Please see this resource: https://vercel.com/support/articles/how-can-i-use-files-in-serverless-functions

**Reply by leerob** [reply]
Alternatively, you can persist the file to Google Sheets: https://leerob.io/snippets/google-sheets

**Comment by dominicfarr** [comment]
@leerob  [leerob.io/snippets/google-sheets](https://leerob.io/snippets/google-sheets) is now a dead link.  Can it still be accessed another way?

I'm using a svelte and would like to include a csv file. 
Running vercel dev locally isn't a problem. 
I cannot figure out how to get this working in vercel production. It's within the vercel production infrastructure where things get moved and hidden or just not included. 
  
The https://vercel.com/guides/how-can-i-use-files-in-serverless-functions#using-sveltekit documentation is sparse on details.  For example, in the import statement `import users from './users.json';` where is `./` is that file location or /static? or root? 

Any help on this would be greatly appreciated. For now I'm wrapping my csv in a js file and importing it that way. Not ideal.

**Reply by leerob** [reply]
You shouldn't need to use `vercel dev` here, you can test locally using the SvelteKit server. The filename is `src/routes/read-file/+page.server.js`, so the CSV file in this instance would be in the same directory, `src/routes/read-file/data.csv`. When you deploy, Vercel will be able to bundle this CSV file so it can be read in production, as well.
